source/extensions/filters/http/cache/cache_policy.h

```diff
@@ -23,82 +22,96 @@ struct CacheEntryUsability {
    * Value to be put in the Age header for cache responses.
    */
   Seconds age = Seconds::max();
+  /**
+   * Remaining freshness lifetime.
+   */
+  Seconds ttl = Seconds::max();
+
+  friend bool operator==(const CacheEntryUsability& a, const CacheEntryUsability& b) {
+    return std::tie(a.status, a.age, a.ttl) == std::tie(b.status, b.age, b.ttl);
+  }
+
+  friend bool operator!=(const CacheEntryUsability& a, const CacheEntryUsability& b) {
+    return !(a == b);
+  }
 };
 
-class CachePolicyCallbacks {
-public:
-  virtual ~CachePolicyCallbacks() = default;
+enum class RequestCacheability {
+  // This request is eligible for serving from cache, and for having its response stored.
+  Cacheable,
+  // Don't respond to this requst from cache, or store its response into cache.
+  Bypass,
+  // This request is eligible for serving from cache, but its response must not be stored.
```

Comment:
Could you elaborate on this comment a bit? I don't understand how you would be able to serve a response from cache if you can't store it.

Reply:
Request 1: "GET /", response gets stored in cache
Request 2: "GET /", but with "cache-control: no-store"

Request 2 can be served from cache, but if it weren't found in cache, we wouldn't[1] store the response.

If all the requests for a particular URL have "cache-control: no-store", then none will ever be stored, but if only some have "no-store", then there may be a cache entry to serve from, even if a particular request can't[1] be stored.

[1] https://httpwg.org/specs/rfc9111.html#cache-request-directive lets caches choose whether to honor request cache-control directives. The products I work on honor "no-store", because that may express a concern that the response could have private information, and that's a concern to always take seriously.

Reply:
Thanks! (Might be worth adding a short note to this effect in the code comment so future people have the same context.)

Also there's a typo requst -> request in the comment above, which I didn't previous mention because CI was mentioning it.